node_modules/oae-core/topnavigation/css/topnavigation.css

```diff
@@ -63,6 +63,12 @@
     }
 }
 
+@media (max-width: 767px) {
+    #topnavigation-right .topnavigation-haveaccount {
+        display: none;
```

Comment:
Ah, there were 2 class elements, and firefox was only honouring one. I've removed the `hidden-sm` from the element, since when the topnav enhancements are merged, all the conditional hiding of elements in the topnav based on context and device width are handled in the `topnavigation.css`